### DISCUSSION TITLE
Is it possible to upload documents such as PDF, DOCX, and TXT files for QA interactions?

### DISCUSSION THREAD

**Discussion by wayfare11** [danny-avila/LibreChat#1581] [Q&A]
### What happened?

Hello, author. I noticed that there is an "ask_for_pdf" feature in the plugin, but I seem to be unable to upload PDFs. Does it support uploading documents for knowledge base Q&A?

### Steps to Reproduce

![image](https://github.com/danny-avila/LibreChat/assets/81129160/47151135-2c02-4b22-a466-da1f7e245b87)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Not yet, it’s being actively developed.

**Reply by JeremyCSUMB** [reply]
Hey Danny, I frequently use LibreChat and greatly appreciate its features and capabilities. However, I am interested in knowing if it could potentially include a feature that allows for the reading of local files from a designated folder. Specifically, this would involve placing PDF files into, or pointing the application to, a specific folder from which it could then read and synthesize information from any of the PDF files contained within. What are your thoughts?

**Comment by natv** [comment]
Is this still the case (not supported yet?) Trying to upload a PDF and not sure if it's something I have to configure somewhere, or just not supported yet.

**Reply by danny-avila** [reply]
It's currently only available through the [Assistants endpoint](https://docs.librechat.ai/install/configuration/ai_setup.html#assistants)

**Reply by Marx1st** [reply]
Is there any documentation on how to use this?

**Reply by simon-77** [reply]
Simply set `ASSISTANTS_API_KEY=user_provided` in the .env file to enable Assistants with a user provided API key.

**Reply by simon-77** [reply]
When you have set your API key:

    Create your own Assistant in LibreChat UI
    select any gpt-4 model (e.g. the cheaper gpt-4-turbo-preview)
    under Capabilities enable Retrieval

And then use this Assistant and attach your file.

**Reply by Marx1st** [reply]
*update*
It suddenly started to work...
*update end*

Thanks! I finally managed to get the assistant endpoint working (writing configuration options correctly is surprisingly relevant... ;) ).

Unfortunatly it doesn't work. I've created an assistant with GPT-4 and retrieval. I create a new chat with it, attach a pdf and ask it to summarize that, but I never get an answer (or error). What am I missing?

**Reply by simon-77** [reply]
> What am I missing?
Hard to tell, for me it worked "out of the box" once I had the *Assistants* configured.

One thing you can do though, is to check the Playground by openAI: https://platform.openai.com/playground
When you log-in there you should see the Assistant you have created in LibreChat and you can test your assistant in the playground to check whether it can read your PDFs there.

Regards, Simon

**Reply by simon-77** [reply]
I also had a situation now where the Assistant responded to my request with a PDF attachement:

> Unfortunately, as the file you uploaded isn't accessible with the tools I have, I cannot refer directly to your microcontroller's datasheet to provide more specific information.

I then responded (**without** attaching the file again):

> It is attached to the chat, please make a file retrieval.

... then it suddenly started retrieving the file.

**Comment by HASH717** [comment]
Hi, I can't create an assistant in the LibreChat UI, I've set ASSISTANTS_API_KEY=user_provided and used librechat.yaml example (uncommented) but still can't upload PDFs, always saying unsupported.

**Reply by danny-avila** [reply]
Do you have an assistant created/selected?

**Reply by HASH717** [reply]
Do I need to create an assistant in OpenAI?

**Comment by karanmgandhi** [comment]
Uploading and QA with PDFs, can we do it normally or we still need to use the Assistant way?

**Reply by danny-avila** [reply]
You can do it now without Assistants

**Comment by Alan-Billi** [comment]
Hello ! 
Are we still not able to upload DOCX or PPTX directly in the chat? 
Is it in the Roadmap or not at all?

Thank you.
